app/services/web_mentions/web_mention_handler.rb

```diff
@@ -0,0 +1,44 @@
+module WebMentions
+  class WebMentionHandler
+    def initialize(canonical_url, article_url = nil, webmention_url = nil)
+      @canonical_url = canonical_url
+      @article_url = article_url
+      @webmention_url = webmention_url
+    end
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def call
+      accepts_webmention? ? send_webmention : Rails.logger.info("#{@canonical_url} doesn't support Webmentions")
+    end
+
+    def accepts_webmention?
+      webmention_url.blank? ? false : true
+    end
+
+    def webmention_url
+      document = Nokogiri::HTML.parse(URI.open(@canonical_url)).at_css("link[rel='webmention']")
+
+      if document
+        @webmention_url = if Addressable::URI.parse(document["href"]).relative?
+                            Addressable::URI.parse(@canonical_url).origin + document["href"]
+                          else
+                            document["href"]
+                          end
+      end
+    rescue StandardError => e
+      Rails.logger.error("WebmentionsException: #{e}")
+      nil
+    end
+
+    private
+
+    def send_webmention
+      RestClient.post(@webmention_url, "source": @article_url, "target": @canonical_url)
```

Comment:
can we switch to `HTTParty.post()` ?

I know we have many HTTP clients but we're trying to converge on actively use only one